File: EducNotes.API/Models/Theme.cs
using System.Collections.Generic;

namespace EducNotes.API.Models
{
  public class Theme
  {
    public int Id { get; set; }
    public int ClassLevelId { get; set; }
    public ClassLevel ClassLevel { get; set; }
    public int CourseId { get; set; }
    public Course Course { get; set; }
    public string Name { get; set; }
    public string Desc { get; set; }
    public ICollection<Lesson> Lessons { get; set; }
  }
}

File: EducNotes-SPA/src/app/_models/theme.ts
import { ClassLevel } from './classLevel';
import { Course } from './course';
import { Lesson } from './lesson';

// import {} from 'class-validator';

export class Theme {
  public id: number;

  public classLevelId: number;

  public classLevel: ClassLevel;

  public courseId: number;

  public course: Course;

  public name: string;

  public desc: string;
  public lessons: Lesson[];

  constructor(
    id?: number,
    classLevelId?: number,
    classLevel?: ClassLevel,
    courseId?: number,
    course?: Course,
    name?: string,
    desc?: string,
  ) {
    this.id = id || 0;
    this.classLevelId = classLevelId || 0;
    this.courseId = courseId || 0;
    this.name = name || '';
    this.desc = desc || '';
  }
}
